module/Database/src/Service/Meeting.php

```diff
@@ -586,27 +587,28 @@ public function installDecision(array $data): array
                 'function' => $value['function'],
             ];
         });
-
         $decision['installations'] = $installations;
 
-        $discharges = [];
-        array_walk($decision['discharges'], function ($value) use (&$discharges): void {
-            $decision = $this->getOrganMapper()->findInstallationDecision(
-                MeetingTypes::from($value['meeting_type']),
-                (int) $value['meeting_number'],
-                (int) $value['decision_point'],
-                (int) $value['decision_number'],
-                (int) $value['number'],
-            );
-
-            if (null === $decision) {
-                return;
-            }
-
-            $discharges[] = $decision;
-        });
-
-        $decision['discharges'] = $discharges;
+        // Prepare reappointments and discharges.
+        foreach (['reappointments', 'discharges'] as $subDecisionType) {
+            $subDecisions = [];
+            array_walk($decision[$subDecisionType], function ($value) use (&$subDecisions): void {
+                $decision = $this->getOrganMapper()->findInstallationDecision(
+                    MeetingTypes::from($value['meeting_type']),
+                    intval($value['meeting_number']),
+                    intval($value['decision_point']),
+                    intval($value['decision_number']),
+                    intval($value['number']),
+                );
+
+                if (null === $decision) {
+                    return;
+                }
+
+                $subDecisions[] = $decision;
```

Comment:
Maybe I misunderstood, but do we want to create discharges for members who are not touched? Currently, no such behaviour occurs, so reappointing "Hugo Herbenoemd" ant not touching "Simon de Afhaker" causes Simon to remain in the organ.

Reply:
My idea was to make the decision(s) very explicit; the secretary _must_ ensure that all decisions are correct and we do not do too much for them behind the scenes (either in the front-end or the back-end).

I am happy to change the front-end to remove the button after reappointing (or newly installing) a member to ensure it does not happen (again)* but I do not want to create the complex logic in the front-end that ensures that newly installed members are not reappointed if it is a body-wide decision (actual decision is still per member but like a 'reappoint members' button instead of one per member). Nor do I want to do this in the back-end as that hides the logic from the secretary.

*I can even do the same for discharges of newly installed members.

Reply:
If this was intended, it is fine